static/app/components/onboarding/gettingStartedDoc/step.tsx

```diff
@@ -195,8 +201,58 @@ export function Step({
   configurations,
   additionalInfo,
   description,
+  isOptional = false,
 }: StepProps) {
-  return (
+  const [showOptionalConfig, setShowOptionalConfig] = useState(false);
+
+  return isOptional ? (
+    <div>
+      <OptionalConfigWrapper>
+        <ToggleButton
+          priority="link"
+          borderless
+          size="zero"
+          icon={<IconChevron direction={showOptionalConfig ? 'down' : 'right'} />}
+          aria-label={t('Toggle optional configuration')}
+          onClick={() => setShowOptionalConfig(!showOptionalConfig)}
+        >
+          <h4 style={{marginBottom: 0}}>
+            {title ?? StepTitle[type]}
+            {t(' (Optional)')}
+          </h4>
+        </ToggleButton>
+      </OptionalConfigWrapper>
+      {showOptionalConfig ? (
+        <Fragment>
+          {description && <Description>{description}</Description>}
+          {!!configurations?.length && (
+            <Configurations>
+              {configurations.map((configuration, index) => {
+                if (configuration.configurations) {
+                  return (
+                    <Fragment key={index}>
+                      {getConfiguration(configuration)}
+                      {configuration.configurations.map(
+                        (nestedConfiguration, nestedConfigurationIndex) => (
+                          <Fragment key={nestedConfigurationIndex}>
+                            {getConfiguration(nestedConfiguration)}
+                          </Fragment>
+                        )
+                      )}
+                    </Fragment>
+                  );
+                }
+                return <Fragment key={index}>{getConfiguration(configuration)}</Fragment>;
+              })}
+            </Configurations>
+          )}
```

Comment:
If this is the same in the 'else' case we should be able to set it to a variable and pass it in.

```
const configurations = configurations?.length ? <Configurations>....</Configurations> : null;

return isOptional ? (
  <div>
    <Togglebutton .... />
    {configurations}
  </div>
) : (
  <div>
    <h4>....</h4>
    {configurations}
  </div>
);
```